test/e2e/middleware-custom-matchers-basepath/test/index.test.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import webdriver from 'next-webdriver'
+import { fetchViaHTTP } from 'next-test-utils'
+import { createNext, FileRef } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+
+describe('Middleware custom matchers basePath', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files: new FileRef(join(__dirname, '../app')),
+    })
+  })
+  afterAll(() => next.destroy())
+
+  it('should match', async () => {
```

Comment:
**suggestion:** for better reporting, you may use `it.each`

```js
it.each([
  { path: '/docs/hello' }, 
  { path: '/docs/about' }
])('$path should match'), async ({ path }) => {
  const res = await fetchViaHTTP(next.url, path)
  expect(res.status).toBe(200)
  expect(res.headers.get('x-from-middleware')).toBeDefined()
})
```

This way, we get a better error reporting because we'll know which path has failed (or succeeded).

Reply:
fixed to use `it.each` including all others